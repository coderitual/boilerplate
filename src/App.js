import React, { Component } from 'react';
import { injectGlobal } from 'styled-components';

injectGlobal`
  body {
    margin: 0;
    padding: 0;
    font-family: sans-serif;
    -webkit-tap-highlight-color: rgba(0,0,0,0);
  }
`;

class App extends Component {
  state = {
    text: 'world',
  };
  render() {
    return (
      <div className="app" onClick={() => this.setState({ text: 'you' })}>
        Hello {this.state.text}
      </div>
    );
  }
}

export default App;
