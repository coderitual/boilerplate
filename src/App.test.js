import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import App from './App';

test('App renders correctly', () => {
  const { container, getByText } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();

  fireEvent.click(getByText('Hello world'));
  expect(container.firstChild).toMatchSnapshot();
});
