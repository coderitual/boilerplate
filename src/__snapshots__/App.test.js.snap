// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  class="app"
>
  Hello 
  world
</div>
`;

exports[`App renders correctly 2`] = `
<div
  class="app"
>
  Hello 
  you
</div>
`;
